src/System.Device.Gpio/System/Device/Gpio/IGpioControllerExtensions.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Disposables;
+
+namespace System.Device.Gpio
+{
+    /// <summary>
+    /// Extensions for <see cref="IGpioController"/> interface.
+    /// </summary>
+    public static class IGpioControllerExtensions
+    {
+        /// <summary>
+        /// Opens a pin in order for it to be ready to use.
+        /// </summary>
+        /// <param name="controller">The GPioController.</param>
+        /// <param name="pinNumber">The pin number in the controller's numbering scheme.</param>
+        /// <returns>A disposable that will close the pin if disposed.</returns>
+        public static IDisposable OpenPinAsDisposable(this IGpioController controller, int pinNumber)
```

Comment:
we're actually thinking of having GpioPin class instead, see https://github.com/dotnet/iot/issues/215